content/blog/2022-07-26-schema-evolution-glue-msk/index.md

```diff
@@ -0,0 +1,861 @@
+```java
+public UnicornRideRequest getRecord(int requestId) {
+    /*
+      Initialise UnicornRideRequest object of
+      class that is generated from AVRO Schema
+    */
+    UnicornRideRequest rideRequest = UnicornRideRequest.newBuilder()
+            .setRequestId(requestId)
+            .setPickupAddress("Melbourne, Victoria, Australia")
+            .setDestinationAddress("Sydney, NSW, Aus")
+            .setRideFare(1200.50F)
+            .setRideDuration(120)
+            .setPreferredUnicornColor(UnicornPreferredColor.WHITE)
+            .setRecommendedUnicorn(RecommendedUnicorn.newBuilder()
+                    .setUnicornId(requestId * 2)
+                    .setColor(unicorn_color.WHITE)
+                    .setStarsRating(5).build())
+            .setCustomer(Customer.newBuilder()
+                    .setCustomerAccountNo(1001)
+                    .setFirstName("Dummy")
+                    .setLastName("User")
+                    .setEmailAddresses(List.of("demo@example.com"))
+                    .setCustomerAddress("Flinders Street Station")
+                    .setModeOfPayment(ModeOfPayment.CARD)
+                    .setCustomerRating(5).build()).build();
+    LOGGER.info(rideRequest.toString());
+    return rideRequest;
+}
+
+private static class ProducerCallback implements Callback {
+    @Override
+    public void onCompletion(RecordMetadata recordMetaData, Exception e) {
+        if (e == null) {
+            LOGGER.info("Received new metadata. \t" +
+                    "Topic:" + recordMetaData.topic() + "\t" +
+                    "Partition: " + recordMetaData.partition() + "\t" +
+                    "Offset: " + recordMetaData.offset() + "\t" +
+                    "Timestamp: " + recordMetaData.timestamp());
+        } else {
+            LOGGER.info("There's been an error from the Producer side");
+            e.printStackTrace();
+        }
+    }
+}
+```
+
+And finally, we can implement our `startProducer` method which creates the producer using `getProducerConfig` and sends records generated by `getRecord`:
+```java
+public void startProducer() {
+    try (KafkaProducer<String, UnicornRideRequest> producer = new KafkaProducer<>(getProducerConfig())) {
+        int numberOfMessages = Integer.parseInt(str_numOfMessages);
+        LOGGER.info("Starting to send records...");
+        for (int i = 0; i < numberOfMessages; i++) {
+            UnicornRideRequest rideRequest = getRecord(i);
+            String key = "key-" + i;
+            ProducerRecord<String, UnicornRideRequest> record = new ProducerRecord<>(topic, key, rideRequest);
+            producer.send(record, new ProducerCallback());
+        }
+    }
+}
+```
+
+[The complete module can be found in our samples repository (along with the rest of the code of this tutorial).](https://github.com/localstack/localstack-pro-samples/blob/cd023a84a3b473984e9c34053d4feb7de8e038c1/glue-msk-schema-registry/producer/)
```

Comment:
maybe we should add the links to the code a the top of the sections, so readers know it's safe to skip the section?